Neos.Neos/Classes/FrontendRouting/EventSourcedFrontendNodeRoutePartHandler.php

```diff
@@ -228,22 +238,25 @@ public function matchWithParameters(&$requestPath, RouteParameters $parameters)
     /**
      * @param string $uriPath
      * @param DimensionSpacePoint $dimensionSpacePoint
-     * @return MatchResult
+     * @return MatchResult|false
```

Comment:
We should add some explainer for our future selves:
```suggestion
     * @return MatchResult|false returns the MatchResult that represents the node that matched the given $uriPath – or false if the found node does not match the constraints
```